### ISSUE TITLE
peformance improvements of mirror synchronisation

### ISSUE THREAD

**Issue by btel** [mamba-org/quetz#196]
* re-generate local indexes before synchronising
* run one or more threads per subdir
* re-use connection to s3
* profiling

**Comment by wolfv** [comment]
1. is fixed now, we just use the information from the database
2. we run a configurable amount of threads to download data, and a configurable amount of threads to extract condainfo & upload data now
3. seems to be fine after we removed the transaction which wasn't multithread safe
4. I have played around with a TicToc statement

Currently the slowest parts are either bandwidth, and CPU power to extract the metadata.

In theory, we could disregard extracting metadata and just use the "upstream"-repodata. However, in the case of conda-forge, the upstream repodata is the patched repodata...

It could also be good to not upload packages in an alphabetically sorted fashion: often similar packages end up in the same batch. The problem is they are usually similarly large leading to the extraction process taking similarly long and we end up with 16 cores at 100% without using the available bandwidth to already upload files.
